src/mpc-hc/MainFrm.cpp

```diff
@@ -6149,13 +6166,37 @@ void CMainFrame::OnFileSubtitlesSave()
             }
 
             // same thing as in the case of CVobSubFile above for lpszDefExt
-            CSaveSubtitlesFileDialog fd(pRTS->m_encoding, m_pCAP->GetSubtitleDelay(), s.bSubSaveExternalStyleFile,
-                                        _T("srt"), suggestedFileName, filter, types, GetModalParent());
+            if (silent) {

```

Comment:
It should check the current subtitle type. SSA/ASS/WebVTT should be saved as ASS. Everything else as SRT.